### DISCUSSION TITLE
writing staggered variables to plotfiles

### DISCUSSION THREAD

**Discussion by BenWibking** [AMReX-Codes/amrex#2904] [Q&A] [answered]
Is it possible to write staggered grid variables to plotfiles with the standard machinery? Or is it necessary to do what WarpX does, e.g.: https://github.com/ECP-WarpX/WarpX/blob/13437f5d7c03e3ab7b6d826497df7351a5a23c66/Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp#L387

We save the face-centered velocities computed within the Riemann solver, and we'd like to write those out in plotfiles and use them for streamline, vorticity, etc., calculations within Visit or yt.

**Comment by atmyers** [comment]
Hi Ben,

AMReX supports writing out staggered variables to plotfiles, but most visualization tools, including Visit and yt, do not fully know how to handle them yet. yt has partial support for staggered variables, explained here: https://yt-project.org/doc/examining/loading_data.html?highlight=staggered#viewing-raw-fields-in-warpx, but in I expect you'd have to modify yt to get it to do exactly what you want.

**Reply by BenWibking** [reply]
Hi Andrew,

Is the best way to do write out staggered variables to copy the WarpX code that does this?

**Comment by atmyers** [comment, marked as answer]
If you'd like to take advantage of the existing capabilities in yt, then yes, you should follow the WarpX code for this.
